velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -16,124 +16,226 @@
 
 #include <utility>
 
+#include "velox/expression/DecodedArgs.h"
 #include "velox/expression/VectorFunction.h"
 #include "velox/expression/VectorWriters.h"
+#include "velox/functions/lib/Re2Functions.h"
 
 namespace facebook::velox::functions::sparksql {
 namespace {
 
-/// This class only implements the basic split version in which the pattern is a
-/// single character
-class SplitCharacter final : public exec::VectorFunction {
+/**
```

Comment:
nit: for multiple-line comment, we need to use `///` for public APIs or `//` otherwise. Here `//` is needed as this class is in an anonymous namespace, seeing https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md#comments.
Comment suggestion:
>// split(string, delimiter[, limit]) -> array(varchar)
//
// Splits string on delimiter and returns an array of size at most limit.
// Delimiter is a string representing regular expression. Limit is an integer
// which controls the number of times the regex is applied. By default, limit is
// -1.